src/arithmetic.jl

```diff
@@ -0,0 +1,501 @@
+# This file is part of the TaylorSeries.jl Julia package, MIT license
+#
+# Luis Benet & David P. Sanders
+# UNAM
+#
+# MIT Expat license
+#
+
+# Arithmetic operations: +, -, *, /
+
+## Equality ##
+for T in (:Taylor1, :TaylorN)
+    @eval begin
+        =={T<:Number,S<:Number}(a::$T{T}, b::$T{S}) = ==(promote(a,b)...)
+        function =={T<:Number}(a::$T{T}, b::$T{T})
+            a, b = fixorder(a, b)
```

Comment:
`fixorder` is needed because you may be comparing two `Taylor1` (or `TaylorN`) polynomials with different order.